### PR TITLE
[Sprint 10 / Feature] Pseudo Label Script

### PR DESCRIPTION
**Pull request by segara2410**
## Jira Link: 

https://ichiro-its.atlassian.net/browse/PI-164

## Description

Add Pseudo label script that can help to label object detection data

## Type of Change

- [ ] Bugfix
- [ ] Enhancement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause the existing functionality to not work as expected)

## How Has This Been Tested?

- [ ] New unit tests added.
- [x] Manual tested.

## Checklist:

- [x] Using Branch Name Convention
    - `feature/JIRA-ID-SHORT-DESCRIPTION` if has a JIRA ticket
    - `enhancement/SHORT-DESCRIPTION` if has/has no JIRA ticket and contain enhancement
    - `hotfix/SHORT-DESCRIPTION` if the change doesn't need to be tested (urgent)
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made the documentation for the corresponding changes.

### CODE REVIEW

File: pseudo_label.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      for image in tqdm(os.listdir(target_dir)):
+        full_path = os.path.join(target_dir, image)
+        if os.path.isfile(full_path) and full_path.split('.')[-1] in ['png', 'jpg', 'jpeg', 'tiff', 'bmp', 'gif']:
+          file_list_text.write(f"{full_path}.txt\n")
+          file_list_text.write(full_path)
+          detected_objects = net.detect(cv2.imread(full_path))
+
+          image_text = open(f"{full_path.split('.')[0]}.txt", "w")
+          for detected_object in detected_objects:
+            image_text.write(f"{detected_object['class_id']} {detected_object['x']} {detected_object['y']} {detected_object['w']} {detected_object['h']}\n")
```
Comment (Nathan24zz) [inline pseudo_label.py]: for better readability, I suggest separating lines that seem too long like this one

File: yolo.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
+from ast import List
+from typing import Tuple
+import cv2
+import numpy as np
+
+class Yolo:
+    def __init__(self, config_path: str, weights_path: str, names_path: str, conf_thresh: float, nms_thresh: float, use_cuda: bool = False) -> None:
+        # save thresholds
+        self.ct = conf_thresh
+        self.nmst = nms_thresh
```
Comment (Nathan24zz) [inline yolo.py]: Better to make the same indentation for all of the files, I think.

File: pseudo_label.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    target_dir = os.path.join(data_path, name)
+    if os.path.isdir(target_dir):
+      print(f"\nRunning on directory {target_dir}...")
+
+      file_list_text = open(f"{target_dir}.txt", "w")
+
+      for image in tqdm(os.listdir(target_dir)):
+        full_path = os.path.join(target_dir, image)
+        if os.path.isfile(full_path) and full_path.split('.')[-1] in ['png', 'jpg', 'jpeg', 'tiff', 'bmp', 'gif']:
+          file_list_text.write(f"{full_path}.txt\n")
```
Comment (Nathan24zz) [inline pseudo_label.py]: I think it will be wrong because it will generate a file that name (file name).(extension file).txt

Comment (segara2410) [reply pseudo_label.py]: updated

File: yolo.py (hunk 2 above)
Comment (segara2410) [reply yolo.py]: updated

File: pseudo_label.py (hunk 1 above)
Comment (segara2410) [reply pseudo_label.py]: updated

**Review by Nathan24zz — APPROVED** [review]
LGTM

**Review by threeal — APPROVED** [review]

**Pull request merged**